Ginger/Ginger/Run/RunSetsExecutionsHistoryPage.xaml.cs

```diff
@@ -360,10 +444,378 @@
+                                    activitiesExistingInSR);
+
+                                ExportUseCaseFromBusinessFlowExecutionSequence(filteredBfSequence);
+                                exportedSuccessfullyCount++;
+                            }
+                            catch (Exception ex)
+                            {
+                                Reporter.ToLog(eLogLevel.ERROR, $"Error occurred while exporting BPMN for business flow {bfSequence.Name}.", ex);
+                            }
+                        }
+                        if (exportedSuccessfullyCount > 0)
+                            Dispatcher.Invoke(() => Reporter.ToUser(eUserMsgKey.MultipleExportToBPMNSuccessful, exportedSuccessfullyCount));
+                        else
+                            Dispatcher.Invoke(() => Reporter.ToUser(eUserMsgKey.GingerEntityToBPMNConversionError, "Unexpected Error, check logs for more details."));
+                        return;
+                    }
+
+                    if (userResponse == eUserMsgSelection.Yes)
+                    {
+                        bool wasAllAdded = TryAddActivitiesToSharedRepository(activitiesMissingFromSR);
+                        if (!wasAllAdded)
+                            return;
+
+                        int exportedSuccessfullyCount = 0;
+                        foreach (BusinessFlowExecutionSequence bfSequence in bfSequences)
+                        {
+                            try
+                            {
+                                ExportUseCaseFromBusinessFlowExecutionSequence(bfSequence);
+                                exportedSuccessfullyCount++;
+                            }
+                            catch (Exception ex)
+                            {
+                                Reporter.ToLog(eLogLevel.ERROR, $"Error occurred while exporting BPMN for business flow {bfSequence.Name}.", ex);
+                            }
+                        }
+                        if (exportedSuccessfullyCount > 0)
+                            Dispatcher.Invoke(() => Reporter.ToUser(eUserMsgKey.MultipleExportToBPMNSuccessful, exportedSuccessfullyCount));
+                        else
+                            Dispatcher.Invoke(() => Reporter.ToUser(eUserMsgKey.GingerEntityToBPMNConversionError, "Unexpected Error, check logs for more details."));
+                        return;
+                    }
+                }
+                catch(Exception ex)
+                {
+                    Dispatcher.Invoke(() => Reporter.ToUser(eUserMsgKey.GingerEntityToBPMNConversionError, "Unexpected Error, check logs for more details."));
+                    Reporter.ToLog(eLogLevel.ERROR, "Exception occurred while exporting BPMN from execution history.", ex);
+                }
+                finally
+                {
+                    Dispatcher.Invoke(() => Reporter.HideStatusMessage());
+                }
+            });
```

Comment:
I agree that this method handles a lot of logic. But since this class is not dedicated for this logic I want to restrict the number of separate methods that I create. Otherwise it will bloat the class with too many methods.